src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c

```diff
@@ -227,19 +230,50 @@ ucc_status_t ucc_tl_mlx5_mcast_coll_setup_comm_resources(ucc_tl_mlx5_mcast_coll_
     comm->pending_recv = 0;
     comm->buf_n        = comm->params.rx_depth * 2;
 
-    ret = posix_memalign((void**) &comm->pp_buf, page_size, buf_size * comm->buf_n);
-    if (ret) {
-        tl_error(comm->ctx->lib, "posix_memalign failed");
-        return UCC_ERR_NO_MEMORY;
+    supported_mem_type = comm->cuda_mem_enabled ? UCC_MEMORY_TYPE_CUDA
+                                                : UCC_MEMORY_TYPE_HOST;
+
+    status = ucc_mc_alloc(&comm->grh_buf_header, GRH_LENGTH *
```

Comment:
this is because i was not sure about the following:

Can i post a scatter-gather list to a receive queue where some of the elements point to GPU memory and some elements point to Host memory?

Since I was unsure, I kept both GRH and data buffers in CPU or GPU.

Reply:
Can you please check that host buffers will work, ucc_mc_alloc takes 1MB buffer from cache and use only 64 bytes of it

Reply:
sure will do

Reply:
yeah it works if I use host memory for grh,  i made the required changes.